File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/AuthorManga.java
package com.amakakeru.mangaworld.Entity;

import com.amakakeru.mangaworld.Entity.IdClasses.AuthorMangaId;
import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;

@Data
@Entity
@IdClass(AuthorMangaId.class)
@Table(name = "author_manga")
public class AuthorManga implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "a_id")
    private Long aId;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "m_id")
    private Long mId;

    @ManyToOne
    @JoinColumn(name = "a_id", insertable = false, updatable = false)
    private Author author;

    @ManyToOne
    @JoinColumn(name = "m_id", insertable = false, updatable = false)
    private Manga manga;
}

File: Frontend/manga-world/src/Model/AuthorManga.ts
import author from "./Author";
import manga from "./Manga";

class authorManga {
    aid: number;
    mid: number;
    author: author;
    manga: manga;

    constructor(aid: number, mid: number, author: author, manga: manga) {
        this.aid = aid;
        this.mid = mid;
        this.author = author;
        this.manga = manga;
    }
}

export default authorManga;